src/frontend/src/optimizer/plan_node/stream_materialize.rs

```diff
@@ -168,14 +168,33 @@ impl StreamMaterialize {
 
         let ctx = input.ctx();
         let properties = ctx.inner().with_options.internal_table_subset();
+        let distribution_key = base.dist.dist_column_indices().to_vec();
+
+        let pk = pk_list.iter().map(|t| t.index).collect_vec();
+        let distribution_key_start_index_in_pk = match distribution_key.is_empty() {
+            true => 0,
```

Comment:
Using `0` as the pk start index when the dist_key is empty seems misleading to me.